diff --git a/meson.build b/meson.build
index b57d9b1..0772a29 100644
--- a/meson.build
+++ b/meson.build
@@ -98,6 +98,7 @@ summary({
             'egl preview' : enable_egl,
             'qt preview' : enable_qt,
             'OpenCV postprocessing' : enable_opencv,
+            'ROS 2 postprocessing' : enable_ros,
             'TFLite postprocessing' : enable_tflite,
             'Hailo postprocessing' : enable_hailo,
             'IMX500 postprocessing' : get_option('enable_imx500'),
diff --git a/meson_options.txt b/meson_options.txt
index 0f24d54..98e2d8b 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -23,6 +23,11 @@ option('enable_opencv',
         value : 'disabled',
         description : 'Enable OpenCV postprocessing support')
 
+option('enable_ros',
+        type : 'feature',
+        value : 'disabled',
+        description : 'Enable ROS 2 postprocessing support')
+
 option('enable_tflite',
         type : 'feature',
         value : 'disabled',
diff --git a/post_processing_stages/meson.build b/post_processing_stages/meson.build
index aa10141..f45377c 100644
--- a/post_processing_stages/meson.build
+++ b/post_processing_stages/meson.build
@@ -68,6 +68,33 @@ if opencv_dep.found()
     enable_opencv = true
 endif
 
+# ROS 2 based postprocessing stages.
+enable_ros = false
+ros_dep = dependency('rclcpp', required : get_option('enable_ros'))
+vision_msgs_dep = dependency('vision_msgs', required : get_option('enable_ros'))
+if ros_dep.found()
+    ros_postproc_src = files([
+        'object_detect_publish_ros_stage.cpp',
+    ])
+
+    # ROS 2 assets
+    postproc_assets += files([
+        assets_dir / 'sobel_cv.json',
+        assets_dir / 'face_detect_cv.json',
+        assets_dir / 'annotate_cv.json',
+    ])
+
+    ros_postproc_lib = shared_module('ros-postproc', ros_postproc_src,
+                                        include_directories : '../..',
+                                        dependencies : [libcamera_dep, ros_dep, vision_msgs_dep],
+                                        cpp_args : cpp_arguments,
+                                        install : true,
+                                        install_dir : posproc_libdir,
+                                        name_prefix : '',
+                                       )
+    enable_ros = true
+endif
+
 # TFlite based postprocessing stages.
 enable_tflite = false
 tflite_dep = dependency('tensorflow-lite', required : get_option('enable_tflite'))
